cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/main/RunMetaFileManager.java

```diff
@@ -76,16 +83,25 @@ public void append(ProgramRunInfo programRunInfo) throws InterruptedException {
                                programRunInfo.getNamespace(), programRunInfo.getTimestamp());
     RunMetaFileOutputStream outputStream = namespaceToLogFileStreamMap.get(programRunInfo.getNamespace());
     appendAndFlushWithRetry(outputStream, programRunInfo);
+    syncOutputStreamsIfRequired();
```

Comment:
We only sync if there are new entries. So say if there are five events

t, t+1sec, t+8secs,t+20secs,t+1hour 

the fourth event will not be seen until the 1 hour event arrives. Instead can we make it sync on a periodic basis?

Reply:
we do call this on periodic basis. as this is also called from https://github.com/caskdata/cdap/blob/fix/CDAP-13758-fixing-latest-status/cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/main/TMSSubscriber.java#L82 which is called when there are no events read from TMS.

Reply:
i will add a comment in the code.